packages/react-components/react-motion/library/src/factories/createPresenceComponent.ts

```diff
@@ -223,11 +221,20 @@ export function createPresenceComponent<MotionParams extends Record<string, Moti
         },
         // Excluding `isFirstMount` from deps to prevent re-triggering the animation on subsequent renders
         // eslint-disable-next-line react-hooks/exhaustive-deps
-        [animateAtoms, handleRef, isReducedMotion, handleMotionFinish, handleMotionStart, handleMotionCancel, visible],
+        [
+          animateAtoms,
+          childRef,
```

Comment:
Should this be `childRef.current`? `childRef` should never change but `.current` might?

Reply:
No, it should not. The only dependency that triggers changes is `visible`.